cmd/ndm_daemonset/probe/usedbyprobe.go

```diff
@@ -200,13 +200,14 @@ func getBlockDeviceZFSPartition(bd blockdevice.BlockDevice) (string, bool) {
 // isBlockDeviceInUseByKernel tries to open the device exclusively to check if the device is
 // being held by some process. eg: If kernel zfs uses the disk, the open will fail
 func isBlockDeviceInUseByKernel(path string) (bool, error) {
-	_, err := os.OpenFile(path, os.O_EXCL, 0444)
+	f, err := os.OpenFile(path, os.O_EXCL, 0444)
 
 	if errors.Is(err, syscall.EBUSY) {
```

Comment:
this just means that its in use by someone, but, need not be by kernel

Reply:
@vishnuitta Whats the best way to check if disk is in use by kernel zfs?

Reply:
how about running zpool commands to find the disks in use by kernel zfs? its also good to change name of the function